Source/Libraries/GSF.Security/OIDCSecurityProvider.cs

```diff
@@ -0,0 +1,582 @@
+        
+        /// <summary>
+        /// Refreshes the <see cref="UserData"/>.
+        /// </summary>
+        /// <returns>true if <see cref="SecurityProviderBase.UserData"/> is refreshed, otherwise false.</returns>
+        public override bool RefreshData()
+        {
+            if (string.IsNullOrEmpty(UserData.Username))
+                return false;
+
+            try
+            {
+                return false;
+            }
+            catch (Exception ex)
+            {
+                LastException = ex;
+                LogError(ex.Source, ex.ToString());
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Authenticates the user.
+        /// </summary>
+        /// <returns>true if the user is authenticated, otherwise false.</returns>
+        public override bool Authenticate()
+        {
+            IsUserAuthenticated = false;
+            TokenResponse token = null;
+            AuthenticationFailureReason = null;
+            Exception authenticationException = null;
+
+            // Test for pre - authentication failure modes such as invalid Tokens
+            if (String.IsNullOrEmpty(UserData.LoginID))
+            {
+                AuthenticationFailureReason = $"No User authorization code is defined.";
+                IsUserAuthenticated = false;
+            }
+
+            if (!String.IsNullOrEmpty(UserData.LoginID))
+                token = GetToken(UserData.LoginID);
+
+            if (token == null)
+            {
+                AuthenticationFailureReason = $"No Token is available.";
+                IsUserAuthenticated = false;
+            }
+
+            // Update the UserData object with new authentication state
+            if (token != null && DecodeToken(token))
+                IsUserAuthenticated = true;
+
+            try
+            {
+                // Log user authentication result
+                LogAuthenticationAttempt(IsUserAuthenticated);
+            }
+            catch (Exception ex)
+            {
+                // Writing data will fail for read-only databases;
+                // all we can do is track last exception in this case
+                LastException = ex;
+                Log.Publish(MessageLevel.Warning, MessageFlags.SecurityMessage, "Authenticate", "Failed to log authentication attempt to database.", "Database or AccessLog table may be read-only or inaccessible.", ex);
+            }
+
+            // If an exception occurred during authentication, rethrow it after logging authentication attempt
+            if ((object)authenticationException != null)
+            {
+                LastException = authenticationException;
+                LogError(authenticationException.Source, authenticationException.ToString());
+                string message = $"Exception occurred during authentication attempt: {authenticationException.Message}";
+                throw new Exception(message, authenticationException);
+            }
```

Comment:
This logic appears to be unnecessary, unless you intended to handle exceptions thrown by `GetToken()` and `DecodeToken()` (which might be a good idea).

```suggestion
```